core/src/main/scala/org/apache/spark/sql/SnappySession.scala

```diff
@@ -239,6 +242,53 @@ class SnappySession(_sc: SparkContext) extends SparkSession(_sc) {
     }
   }
 
+  val unresolvedStarRegex = """(cannot resolve ')(\w+).(\w+).*(' give input columns.*)""".r
+  val unresolvedColRegex = """(cannot resolve '`)(\w+).(\w+).(\w+)(.*given input columns.*)""".r
+
+  // Hack to fix SNAP-2440 ( TODO: Will return after 102 to see if a better fix is warranted )
+  private def reAnalyzeForUnresolvedAttribute(
+    originalPlan: LogicalPlan, e: AnalysisException,
+    retryCount: Int): Option[LogicalPlan] = {
+    if (!e.getMessage().contains("cannot resolve")) return None
+    val unresolvedNodes = originalPlan.expressions.filter(
+      x => x.isInstanceOf[UnresolvedStar] | x.isInstanceOf[UnresolvedAttribute])
+    if (retryCount > unresolvedNodes.size) return None
+
+    val oneLineMsg = e.getMessage().split('\n').map(_.trim.filter(_ >= ' ')).mkString
+    val newPlan = originalPlan transformExpressions {
+      case us@UnresolvedStar(option) if (option.isDefined) =>
+        val targetString = option.get.mkString(".")
+        var matched = false
+        oneLineMsg match {
+          case unresolvedStarRegex(first, schema, table, last) =>
+            if (sessionCatalog.tableExists(new QualifiedTableName(schema, table))) {
+              val qname = s"$schema.$table"
+              if (qname.equalsIgnoreCase(targetString)) {
+                matched = true
+              }
+            }
+          case _ => matched = false
+        }
+        if (matched) UnresolvedStar(None) else us
```

Comment:
This looks incorrect. What if there are multiple qualified names like "app.table1.\*, app.table2.\*"?

Reply:
@sumwale If there are multiple such comma separated elements then each will have their own UnresolvedAttribute node. If you see the logic the retry is tried for number of such nodes present in the plan. I just tested it and it works fine.
This is what I tried:
select a.b.*, a.c.* from a.b, a.c;
Any other syntax you are worried about?